dashboard/test/lib/dcdo_test.rb

```diff
@@ -19,7 +20,20 @@ class DCDOTest < ActiveSupport::TestCase
     }
     key = 'random'
     DCDO.set(key, to_store)
-    assert_equal DCDO.get(key, nil), to_store
```

Comment:
The test "basic set and get" is quite similar, is it also flaky?
https://github.com/code-dot-org/code-dot-org/blob/a6859c54dd045d99396375ab85c11388452141d0/dashboard/test/lib/dcdo_test.rb#L6-L10

Reply:
I have a suspicion that the issue might be in `DCDO.stubs(:get)`, which we use frequently in other test cases.

We could try unstubbing the method `DCDO.get` before each test:
```ruby
setup do
  DCDO.unstub(:get)
end

test 'basic set and get' do 
  assert_nil DCDO.get('key', nil) 
  DCDO.set('key', 'okay') 
  assert_equal DCDO.get('key', nil), 'okay' 
end 
```

or test a new instance of `DCDO` that is not stubbed:
```ruby
setup do
  @dcdo = DCDO.dup
end

test 'basic set and get' do
  assert_nil @dcdo.get('key', nil) 
  @dcdo.set('key', 'okay') 
  assert_equal @dcdo.get('key', nil), 'okay' 
end
```

Reply:
We could also ensure that the `DCDO` will not be rewritten in any other place by using a unique key for each test:
```ruby
test 'basic set and get' do
  dcdo_key = SecureRandom.hex
  dcdo_value = 'okay'

  assert_changes -> {DCDO.get(dcdo_key, nil)}, from: nil, to: dcdo_value do
    DCDO.set(dcdo_key, dcdo_value)
  end
end
```

Reply:
> The test "basic set and get" is quite similar, is it also flaky?
> 
> https://github.com/code-dot-org/code-dot-org/blob/a6859c54dd045d99396375ab85c11388452141d0/dashboard/test/lib/dcdo_test.rb#L6-L10

Good catch - I'm assuming it also has the potential to be flaky even though it hasn't been called out yet, so I added the retry there as well.

Reply:
> I have a suspicion that the issue might be in `DCDO.stubs(:get)`, which we use frequently in other test cases.
> 

Good idea; I added the unstub